AmazonChimeSDKDemo/AmazonChimeSDKDemo/LiveTranscriptionOptionsViewController.swift

```diff
@@ -139,31 +223,150 @@ class LiveTranscriptionOptionsViewController: UIViewController, UITextFieldDeleg
+        piiContentRedactionPickerView.dataSource = self
         
         enginePickerView.tag = 1
         languagePickerView.tag = 2
         regionPickerView.tag = 3
+        partialResultsStabilizationPickerView.tag = 4
+        piiContentIdentificationPickerView.tag = 5
+        piiContentRedactionPickerView.tag = 6
+        
+        storyBoard = UIStoryboard(name: "Main", bundle: nil)
+        languageDetectionViewController = storyBoard!.instantiateViewController(withIdentifier: "languageDetection") as? AutomaticLanguageDetectionViewController
+        
+        languageDetectionViewController?.cancellationHandler = { numberOfselectedLanguages in
+            self.enableAutomaticLanguageDetectionSwitch.isOn = numberOfselectedLanguages! > 1
+        }
+        
+        languageDetectionViewController?.selectedLanguageOptionsHandler = { selectedLanguages in
+            self.selectedLanguagesDetectionOptions = selectedLanguages!
+        }
+        
+        languageDetectionViewController?.preferredLanguageOptionHandler = { preferredLanguage in
+            self.preferredLanguagesDetectionOptions = preferredLanguage!
+        }
+        
+        self.languageHandler = { languageList in
+            self.languageDetectionViewController?.languages = languageList!
+        }
+        
+        self.languagesDictHandler = { mappings in
+            self.languageDetectionViewController?.languagesDict = mappings!
+        }
     }
     
     @IBAction func startTranscriptionButton(_ sender: UIButton) {
-        let url = self.meetingEndpointUrl.hasSuffix("/") ? self.meetingEndpointUrl : "\(self.meetingEndpointUrl)/"
+        var url = AppConfiguration.url
+        url = url.hasSuffix("/") ? url : "\(url)/"
+        
+        let contentIdentificationEnabled = piiContentIdentificationSelected != ""
+        let contentRedactionEnabled = piiContentRedactionSelected != ""
+        
+        var identifyLanguage: Bool?
+        var enablePartialResultsStabilization: Bool?
+        var contentIdentificationType: String?
+        var contentRedactionType: String?
+        var partialResultsStabilty: String?
+        var piiEntityTypes: String?
+        var languageModelName: String?
+        var languageOptions: String?
+        var preferredLanguage: String?
+        
+        if contentIdentificationEnabled {
+            if engineSelected == "transcribe_medical" {
+                contentIdentificationType = "PHI"
+            } else {
+                piiEntityTypes = piiContentIdentificationSelected
+                contentIdentificationType = "PII"
+            }
+        }
+        
+        if contentRedactionEnabled {
+            if engineSelected == "transcribe_medical" {
+                contentRedactionType = "PHI"
+            } else {
+                piiEntityTypes = piiContentRedactionSelected
+                contentRedactionType = "PII"
+            }
+        }
+        
+        if partialResultsStabilizationSelected > 0 {
```

Comment:
the variable name is little confusing to me. Why is it a count, but not a bool?

Reply:
that's the index of the spinner, low, Medium, High, I'll change the name `partialResultsStabilizationOption`